From e040b16df036801a35c3d5ffed9de7452f57d773 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Wed, 6 Dec 2017 23:55:51 +0800
Subject: [PATCH 2999/5242] MLK-17092 staging: typec: enable vbus voltage low
 alarm

commit  d4f36fbd261bd716895974f4452cc98bf279ea40 from
https://source.codeaurora.org/external/imx/linux-imx.git

We use vbus low voltage alarm to start vbus discharge to meet
timing requirement on turning off vbus for power swap from
source to sink, per type-C port controller spec(tcpci), the
Voltage Alarms Power status reporting is disabled by default,
so we need enable it at tcpci init.

Acked-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/staging/typec/tcpci.c |    5 +++++
 drivers/staging/typec/tcpci.h |    1 +
 2 files changed, 6 insertions(+)

diff --git a/drivers/staging/typec/tcpci.c b/drivers/staging/typec/tcpci.c
index bfb649e..572d0ff 100644
--- a/drivers/staging/typec/tcpci.c
+++ b/drivers/staging/typec/tcpci.c
@@ -542,6 +542,11 @@ static int tcpci_init(struct tcpc_dev *tcpc)
 	if (ret < 0)
 		return ret;
 
+	/* Enable Voltage Alarms Power status reporting */
+	regmap_read(tcpci->regmap, TCPC_POWER_CTRL, &reg);
+	reg &= ~TCPC_POWER_CTRL_DIS_VOL_ALARM;
+	ret = regmap_write(tcpci->regmap, TCPC_POWER_CTRL, reg);
+
 	reg = TCPC_ALERT_TX_SUCCESS | TCPC_ALERT_TX_FAILED |
 		TCPC_ALERT_TX_DISCARDED | TCPC_ALERT_RX_STATUS |
 		TCPC_ALERT_RX_HARD_RST | TCPC_ALERT_CC_STATUS |
diff --git a/drivers/staging/typec/tcpci.h b/drivers/staging/typec/tcpci.h
index 7ac11f2..81e2046 100644
--- a/drivers/staging/typec/tcpci.h
+++ b/drivers/staging/typec/tcpci.h
@@ -60,6 +60,7 @@
 #define TCPC_POWER_CTRL			0x1c
 #define TCPC_POWER_CTRL_VCONN_ENABLE	BIT(0)
 #define TCPC_POWER_CTRL_FORCEDISCH	BIT(2)
+#define TCPC_POWER_CTRL_DIS_VOL_ALARM	BIT(5)
 
 #define TCPC_CC_STATUS			0x1d
 #define TCPC_CC_STATUS_TOGGLING		BIT(5)
-- 
1.7.9.5

